### PR TITLE
Fix typo in rustc_driver::version

### PR DESCRIPTION
**Pull request by bjorn3**
This caused rustc `-Zcodegen-backend=foo.so -vV` to look for `oo.so` instead of `foo.so`

### CODE REVIEW

**Review by wesleywiser — APPROVED** [review]

File: compiler/rustc_driver/src/lib.rs [hunk 1]
```diff
@@ -766,7 +766,7 @@ pub fn version(binary: &str, matches: &getopts::Matches) {
         let debug_flags = matches.opt_strs("Z");
         let backend_name = debug_flags.iter().find_map(|x| {
             if x.starts_with("codegen-backend=") {
-                Some(&x["codegen-backends=".len()..])
+                Some(&x["codegen-backend=".len()..])
             } else {
                 None
             }
```
Comment (cuviper) [inline compiler/rustc_driver/src/lib.rs:772]: This expression is the same as `x.strip_prefix("codegen-backend=")`.

Comment (bjorn3) [reply compiler/rustc_driver/src/lib.rs]: Changed to `strip_prefix`.

**Pull request merged**